test/libgit2.jl

```diff
@@ -1073,6 +1082,47 @@ mktempdir() do dir
         end
     end
 
+    @testset "Modify remote" begin
+        repo = LibGit2.GitRepo(test_repo)
+        try
+            name = "test"
+            url = "https://test.com/repo"
+
+            @test !is_defined_remote(repo, name)
+
+            # Set just the fetch URL
+            LibGit2.set_remote_fetch_url(repo, url, name)
```

Comment:
this is testing the `repo::GitRepo` versions pretty well, and the existing block was testing some of the `path::AbstractString` versions, but not the ones you're newly adding